modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php

```diff
@@ -26,8 +38,120 @@ public function onTerminateResponse(KernelEvent $event) {
+  }
+
+  /**
+   * Sends an event to Amplitude.
+   *
+   * @param array $event
+   *   The Amplitude event.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *
+   * @see https://developers.amplitude.com/#http-api
+   */
+  private function sendEvent(array $event) {
+    $this->httpClient->request('POST', $this->apiUrl, [
+      'form_params' => [
+        'api_key' => $this->getApiKey(),
+        'event' => Json::encode($event),
+      ],
+    ]);
+  }
+
+  /**
+   * Creates and sends an event to Amplitude.
+   *
+   * @param string $event_type
+   *   The event type. This accepts any string that is not reserved. Reserved
+   *   event types include: "[Amplitude] Start Session", "[Amplitude] End
+   *   Session", "[Amplitude] Revenue", "[Amplitude] Revenue (Verified)",
+   *   "[Amplitude] Revenue (Unverified)", and "[Amplitude] Merged User".
+   * @param array $event_properties
+   *   (optional) Event properties.
+   *
+   * @return bool
+   *   TRUE if event was successfully sent, otherwise FALSE.
+   *
+   * @throws \Exception
+   *   Thrown if state key acquia_telemetry.loud is TRUE and request fails.
+   *
+   * @see https://amplitude.zendesk.com/hc/en-us/articles/204771828#keys-for-the-event-argument
+   */
+  public function sendTelemetry($event_type, array $event_properties = []) {
+    if ($this->state->get('acquia_cms_telemetry.telemetry') === NULL) {
+      $event = $this->sendTelemetryDataToAmplitude();
+    }
+
+    // Failure to send Telemetry should never cause a user facing error or
+    // interrupt a process. Telemetry failure should be graceful and quiet.
+    try {
+      $this->sendEvent($event);
+      return TRUE;
+    }
+    catch (\Exception $e) {
+      if ($this->state->get('acquia_connector.telemetry.loud')) {
+        throw new \Exception($e->getMessage(), $e->getCode(), $e);
+      }
+      return FALSE;
+    }
+  }
+
+  /**
+   * Send telemetry data to amplitude.
+   */
+  private function sendTelemetryDataToAmplitude() {
+    if (AcquiaDrupalEnvironmentDetector::isAhEnv()) {
+      $appUuid = AcquiaDrupalEnvironmentDetector::getAhApplicationUuid();
+      $siteGroup = AcquiaDrupalEnvironmentDetector::getAhGroup();
+      $env = AcquiaDrupalEnvironmentDetector::getAhEnv();
+      $starterKitName = $this->state->get('acquia_cms.starter_kit');
+      $starterKitUi = $this->state->get('starter_kit_wizard_completed', FALSE);
+      $this->state->set('acquia_cms_telemetry.telemetry', TRUE);
+      return [
+        'application_uuid' => $appUuid,
+        'application_name' => $siteGroup,
+        'environment_name' => $env,
+        'starter_kit_name' => $starterKitName,
+        'starter_kit_ui' => $starterKitUi,
+        'site_studio_status' => $this->siteStudioStatus(),
+      ];
+    }
+  }
+
+  /**
+   * Get site studio status.
+   */
+  public function siteStudioStatus() {
+    if (\Drupal::getContainer()->has('cohesion.utils') &&
```

Comment:
@deepakmishra2 can we Implement this (Container and services) with D.I?

Reply:
@apathak18 no we cannot implement DI here, because there might be scenarios when we wont have cohesion for other starter kits, in that case we will get error.